### ISSUE TITLE
How do I get reshard.pt to run OPT175B?

### ISSUE THREAD

**Issue by awkrail** [facebookresearch/metaseq#574]
### Before asking:   
1. search the issues.   
2. search the docs.    

#### What is your question?
TL;DR: Although I run the `metaseq.scripts.reshard_fsdp` scripts, I could not find `reshard.pt`, `dict.txt`, `gpt2-merges.txt`, and `gpt2-vocab.json`. They are necessary to run `metaseq-api-local`. How do I get them?

#### What have you tried?
I downloaded OPT175B by a given link and run the following `metaseq.scripts.reshard_fsdp` scripts:
```
for j in {0..7}; do
    python -m metaseq.scripts.reshard_fsdp \
    --input-glob-pattern "/path/to/raw/checkpoints/checkpoint_last-model_part-$j-shard*.pt" \
    --output-shard-name "/path/to/resharded/checkpoints/reshard-model_part-$j.pt" \
    --num-output-shards 1 --skip-optimizer-state True --unflatten-weights True
done
```
This command generates `reshard-model_part-$j.pt` in ""/path/to/resharded/checkpoints/". I confirmed it like:
```
[acb11901mk@es1 checkpoints]$ ls
reshard-model_part-2.pt  reshard-model_part-6.pt
reshard-model_part-3.pt  reshard-model_part-7.pt
reshard-model_part-0.pt  reshard-model_part-4.pt
reshard-model_part-1.pt  reshard-model_part-5.pt
```
However, there are no files like `reshard.pt`, `dict.txt`, `gpt2-merges.txt`, and `gpt2-vocab.json`...
When I run `metaseq-api-local`, I got an error as:
```
(opt175b) [acb11901mk@a0090 metaseq]$ metaseq-api-local
2022-12-29 10:15:53 | INFO | metaseq.hub_utils | loading model(s) from /groups/gcb50205/actx/opt_175b_parallel/checkpoints/reshard.pt
Traceback (most recent call last):
  File "/home/acb11901mk/opt175b/bin/metaseq-api-local", line 8, in <module>
    sys.exit(cli_main())
  File "/home/acb11901mk/metaseq/metaseq/cli/interactive_hosted.py", line 389, in cli_main
    distributed_utils.call_main(cfg, worker_main, namespace_args=args)
  File "/home/acb11901mk/metaseq/metaseq/distributed/utils.py", line 290, in call_main
    return main(cfg, **kwargs)
  File "/home/acb11901mk/metaseq/metaseq/cli/interactive_hosted.py", line 188, in worker_main
    models = generator.load_model()  # noqa: F841
  File "/home/acb11901mk/metaseq/metaseq/hub_utils.py", line 169, in load_model
    models, _model_args, _task = _load_checkpoint()
  File "/home/acb11901mk/metaseq/metaseq/hub_utils.py", line 152, in _load_checkpoint
    return checkpoint_utils.load_model_ensemble_and_task(
  File "/home/acb11901mk/metaseq/metaseq/checkpoint_utils.py", line 489, in load_model_ensemble_and_task
    state = load_checkpoint_to_cpu(filename, arg_overrides)
  File "/home/acb11901mk/metaseq/metaseq/checkpoint_utils.py", line 423, in load_checkpoint_to_cpu
    paths_to_load = get_paths_to_load(path, suffix="shard")
  File "/home/acb11901mk/metaseq/metaseq/checkpoint_utils.py", line 377, in get_paths_to_load
    if not _is_checkpoint_sharded(checkpoint_files):
  File "/home/acb11901mk/metaseq/metaseq/checkpoint_utils.py", line 319, in _is_checkpoint_sharded
    sd = torch_load_cpu(checkpoint_files[0])
  File "/home/acb11901mk/metaseq/metaseq/file_io/__init__.py", line 57, in torch_load_cpu
    state = torch.load(path, map_location=torch.device("cpu"))
  File "/home/acb11901mk/opt175b/lib/python3.10/site-packages/torch/serialization.py", line 699, in load
    with _open_file_like(f, 'rb') as opened_file:
  File "/home/acb11901mk/opt175b/lib/python3.10/site-packages/torch/serialization.py", line 230, in _open_file_like
    return _open_file(name_or_buffer, mode)
  File "/home/acb11901mk/opt175b/lib/python3.10/site-packages/torch/serialization.py", line 211, in __init__
    super(_open_file, self).__init__(open(name, mode))
FileNotFoundError: [Errno 2] No such file or directory: '/groups/gcb50205/actx/opt_175b_parallel/checkpoints/reshard.pt'
```

#### What's your environment?

 - metaseq Version (e.g., 1.0 or master): master
 - PyTorch Version (e.g., 1.0): 1.12.1
 - OS (e.g., Linux): Red Hat Enterprise Linux
 - How you installed metaseq (`pip`, source):
 - Build command you used (if compiling from source):
 - Python version: 3.10.4
 - CUDA/cuDNN version: CUDA:11.3 cuDNN: 8.6 
 - GPU models and configuration: NVIDIA A100 40GB * 8
 - Any other relevant information:

**Comment by tangbinh** [comment]
@misogil0116 Indeed, the `metaseq.scripts.reshard_fsdp` script isn't supposed to create any `reshard.pt` file. The suffix `model_part-2*.pt` will be automatically added to `reshard.pt` if the distributed process groups are initialized properly. Can you please share the `LAUNCH_ARGS` in your `constants.py`? Do you have Slurm installed? See the discussion in #407 for some relevant information.

**Comment by tangbinh** [comment]
It's been a while without any update, so we're closing the issue now. @awkrail Please reopen it if you need further support.

**Comment by awkrail** [comment]
Thanks. I understood that my hardwares do not meet the OPT175-B's requirements.
